src/containerapp/azext_containerapp/containerapp_env_decorator.py

```diff
@@ -345,6 +347,26 @@ def post_process(self, r):
 
 class ContainerappEnvPreviewCreateDecorator(ContainerAppEnvCreateDecorator):
 
+    def construct_payload(self):
+        super().construct_payload()
+        if self.get_argument_enable_workload_profiles() and self.get_argument_infrastructure_subnet_resource_id() is not None:
+            self.set_up_infrastructure_resource_group()
+
+    def validate_arguments(self):
+        super().validate_arguments()
+
+        # Infrastructure Resource Group
+        if self.get_argument_infrastructure_resource_group() is not None:
+            if not self.get_argument_infrastructure_subnet_resource_id():
+                raise RequiredArgumentMissingError("Cannot use --infrastructure-resource-group/-i without "
+                                                   "--infrastructure-subnet-resource-id/-s")
+            if not self.get_param("enable_workload_profiles"):
```

Comment:
please use `self.get_argument_enable_workload_profiles()` to get argument value.

Reply:
Done.